rootfs/api/models/__init__.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""
+Data models for the Deis API.
+"""
+
+from __future__ import unicode_literals
+import etcd
+import logging
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+from rest_framework.authtoken.models import Token
+
+
+logger = logging.getLogger(__name__)
+
+
+def log_event(app, msg, level=logging.INFO):
+    # controller needs to know which app this log comes from
+    logger.log(level, "{}: {}".format(app.id, msg))
+    app.log(msg, level)
+
+
+def get_etcd_client():
+    if not hasattr(get_etcd_client, "client"):
+        # wire up etcd publishing if we can connect
+        try:
+            get_etcd_client.client = etcd.Client(
+                host=settings.ETCD_HOST,
+                port=int(settings.ETCD_PORT)
+            )
+
+            get_etcd_client.client.get('/deis')
+        except etcd.EtcdException:
+            logger.warn('Cannot synchronize with etcd cluster')
+            get_etcd_client.client = None
+
+    return get_etcd_client.client
+
+
+# define update/delete callbacks for synchronizing
+# models with the configuration management backend
+
+def _etcd_purge_user(**kwargs):
+    try:
+        url = '/deis/builder/users/{}'.format(kwargs['instance'].username)
```

Comment:
not sure what the old code did, but does this need a check to see if `'instance'` exists in the dict?